### PR TITLE
MG-2414 -  Move auth to middleware in Users service

### LINKED ISSUE #2414: Feature: Move auth to middleware in users service
### Is your feature request related to a problem? Please describe.

Related to issue #2028 

### Describe the feature you are requesting, as well as the possible use case(s) for it.

Move exixting auth implementation to middleware for the users service

### Indicate the importance of this feature to you.

Must-have

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by felixgateru**
# What type of PR is this?
This is a feature as it moves the existing auth to midleware.

## What does this do?
This pr creates the following files: 
 -  users/middleware/auth.go
 and modifies the following:
 - users/service.go

## Which issue(s) does this PR fix/relate to?

- Resolves #2414

## Did you document any new/modified feature?
Yes, in code documentation of fucntions has been updated.

### CODE REVIEW

File: users/api/endpoint_test.go [hunk 1]
```diff
@@ -224,7 +224,7 @@ func TestRegisterClient(t *testing.T) {
 			body:        strings.NewReader(data),
 		}
 
-		svcCall := svc.On("RegisterClient", mock.Anything, tc.token, tc.client).Return(tc.client, tc.err)
+		svcCall := svc.On("RegisterClient", mock.Anything, mock.Anything, tc.client, true).Return(tc.client, tc.err)
```
Comment (dborovcanin) [inline users/api/endpoint_test.go]: Please don't forget to revert this once we update the svc interface.

File: pkg/auth/auth.go [hunk 2]
```diff
@@ -7,9 +7,28 @@ import (
...
 
+type Token struct {
+	AccessToken  string // AccessToken contains the security credentials for a login session and identifies the client.
+	RefreshToken string // RefreshToken is a credential artifact that OAuth can use to get a new access token without client interaction.
+	AccessType   string // AccessType is the specific type of access token issued. It can be Bearer, Client or Basic.
+	UserID       string
+	DomainID     string
+	Name         string
+	Role         clients.Role
+}
```
Comment (dborovcanin) [inline pkg/auth/auth.go]: Why don't we stick to the Magistrala token?

File: pkg/auth/auth.go [hunk 3]
```diff
@@ -7,9 +7,28 @@ import (
...
+
+type AuthObject struct {
+	ID         string
+	UserID     string
+	DomainID   string
+	Token      string
+	SuperAdmin bool
+}
```
Comment (dborovcanin) [inline pkg/auth/auth.go]: Let's rename AuthObject to Session. Do we need a Token string in it? What's the ID for?

**Review by dborovcanin — CHANGES_REQUESTED** [review]

File: pkg/auth/auth.go (hunk 2 above)
Comment (felixgateru) [reply pkg/auth/auth.go]: We require the additional fields not in the magistrala token or do I add the fields to the magistrala token?

**Review by dborovcanin — APPROVED** [review]

**Pull request merged**